### PR TITLE
Fixed issue #18473: Survey listing : multiple call of template

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: avoid multiple call of TemplateConfiguration::model()->search()

### CODE REVIEW

**Comment by Shnoulle** [comment]
Clearly need a better solution here : 

1. no need to call _search_ to have only the list of templet
2. need only the template name from DB

**Comment by Shnoulle** [comment]
Ok, better fix now !

File: application/models/Template.php [hunk 1]
```diff
@@ -470,23 +467,32 @@ public static function getTemplatesFileFolder($sTemplateName = "")
...
+            return $aTemplateList;
+        }
         $aTemplateList = array();
-
-        $oTemplateList = TemplateConfiguration::model()->search();
-        $oTemplateList->setPagination(false);
-
-        foreach ($oTemplateList->getData() as $oTemplate) {
-            $aTemplateList[$oTemplate->template_name] =  (self::isStandardTemplate($oTemplate->template_name)) ?  Yii::app()->getConfig("standardthemerootdir") . DIRECTORY_SEPARATOR . $oTemplate->template->folder : Yii::app()->getConfig("userthemerootdir") . DIRECTORY_SEPARATOR . $oTemplate->template->folder;
+        $aTemplateList = [];
```
Comment (lajosarpad) [inline application/models/Template.php:481]: Why do you create this array twice?

**Review by lajosarpad — CHANGES_REQUESTED** [review]
Overall this looks good, we have a duplicated variable initialization though and we will need to get rid of that

File: application/models/Template.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/Template.php:481]: Error :)

**Review by lajosarpad — APPROVED** [review]

**Review by c-schmitz — APPROVED** [review]

**Pull request merged**